File: util/log/logger.py
import sys
from typing import List, Set, ClassVar
from util.log.logger_print_mode import LoggerPrintMode

class Logger:
    """
    A simple lightweight logger for LAPRAS.
    simpler alternative to full logging.
    """
    
    _instance = None
    _print_modes: Set[LoggerPrintMode] = {LoggerPrintMode.INFO, LoggerPrintMode.ERROR, LoggerPrintMode.DEBUG}  # default
    
    @classmethod
    def set_mode(cls, mode_conf: List[LoggerPrintMode]) -> None:
        """
        Set which logging levels should be displayed.
        
        Args:
            mode_conf: List of enabled logging levels
        """
        if not cls._instance:
            cls._instance = Logger()
        
        cls._print_modes = set(mode_conf)
    
    @classmethod
    def info(cls, log: str) -> None:
        """
        Log an info message if INFO level is enabled.
        
        Args:
            log: The message to log
        """
        if not cls._instance:
            cls._instance = Logger()
        
        if cls._allowed(LoggerPrintMode.INFO):
            print(f"[INFO] {log}")
    
    @classmethod
    def debug(cls, log: str) -> None:
        """
        Log a debug message if DEBUG level is enabled.
        
        Args:
            log: The message to log
        """
        if not cls._instance:
            cls._instance = Logger()
        
        if cls._allowed(LoggerPrintMode.DEBUG):
            print(f"[DEBUG] {log}")
    
    @classmethod
    def error(cls, log: str) -> None:
        """
        Log an error message if ERROR level is enabled.
        
        Args:
            log: The message to log
        """
        if not cls._instance:
            cls._instance = Logger()
        
        if cls._allowed(LoggerPrintMode.ERROR):
            print(f"[ERROR] {log}", file=sys.stderr)
    
    @classmethod
    def _allowed(cls, q_mode: LoggerPrintMode) -> bool:
        """
        Check if a logging level is enabled.
        
        Args:
            q_mode: The logging level to check
            
        Returns:
            True if the logging level is enabled, False otherwise
        """
        return q_mode in cls._print_modes

File: lapras-middleware/src/main/java/kr/ac/kaist/cdsn/lapras/util/log/Logger.java
package kr.ac.kaist.cdsn.lapras.util.log;

/**
 * Basically, every Lapras agent prints log based on slf4j. However, since slf4j library has several dependencies,
 * it is not a proper method to write logs in a light-weight manner. kr.ac.kaist.cdsn.lapras.util.log.Logger is a
 * lighter logging method which can be leveraged for local testing/debugging purposes.
 *
 * Created by chad1231 on 12/01/2017.
 */
public class Logger {
	private static Logger _instance;
	private static LoggerPrintMode[] PRINT_MODE;

	private Logger(){
		PRINT_MODE = new LoggerPrintMode[] {LoggerPrintMode.INFO, LoggerPrintMode.ERROR, LoggerPrintMode.DEBUG}; //default
	}

	public static void setMode(LoggerPrintMode[] modeConf){
		if(_instance == null){
			_instance = new Logger();
		}

		_instance.PRINT_MODE = new LoggerPrintMode[modeConf.length];

		for(int i=0; i<modeConf.length; i++){
			PRINT_MODE[i] = modeConf[i];
		}
	}

	public static void info(String log){
		if(_instance == null){
			_instance = new Logger();
		}

		if(allowed(LoggerPrintMode.INFO)){
			System.out.println(log);
		}
	}

	public static void debug(String log){
		if(_instance == null){
			_instance = new Logger();
		}

		if(allowed(LoggerPrintMode.DEBUG)){
			System.out.println(log);
		}
	}

	public static void error(String log){
		if(_instance == null){
			_instance = new Logger();
		}

		if(allowed(LoggerPrintMode.ERROR)){
			System.out.println(log);
		}
	}

	private static boolean allowed(LoggerPrintMode qMode){
		for(LoggerPrintMode allowedMode : _instance.PRINT_MODE){
			if(allowedMode == qMode) return true;
		}

		return false;
	}
}
